File: guides/micronaut-security-api-key/java/src/main/java/example/micronaut/ApiController.java
package example.micronaut;

import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import io.micronaut.security.annotation.Secured;
import io.micronaut.security.rules.SecurityRule;

import java.security.Principal;

@Controller("/api") // <1>
class ApiController {

    @Produces(MediaType.TEXT_PLAIN) // <2>
    @Get // <3>
    @Secured(SecurityRule.IS_AUTHENTICATED) // <4>
    String index(Principal principal) { // <5>
        return "Hello " + principal.getName();
    }
}

File: guides/micronaut-security-api-key/python/src/example/micronaut/api_controller.py
from java.security import Principal
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


# tag::clazz[]
@Produces(MediaType.TEXT_PLAIN)  # <2>
@Secured(SecurityRule.IS_AUTHENTICATED)  # <4>
@Get("/api")  # <1> <3>
def index(principal: Principal) -> str:  # <5>
    return f"Hello {principal.getName()}"
# end::clazz[]
